test/commands/auth/logout.nut.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { execCmd, TestSession } from '@salesforce/cli-plugins-testkit';
+import { expect } from 'chai';
+import { Env } from '@salesforce/kit';
+import { ensureString, getString } from '@salesforce/ts-types';
+
+let testSession: TestSession;
+
+describe('auth:logout NUTs', () => {
+  const env = new Env();
+
+  before('ensure required environment variables exist', () => {
+    ensureString(env.getString('TESTKIT_JWT_KEY'));
+    ensureString(env.getString('TESTKIT_JWT_CLIENT_ID'));
+    ensureString(env.getString('TESTKIT_HUB_INSTANCE'));
+    ensureString(env.getString('TESTKIT_HUB_USERNAME'));
+  });
+
+  beforeEach(() => {
+    testSession = TestSession.create({});
+  });
+
+  afterEach(async () => {
+    await testSession.clean();
+  });
+
+  it('should remove the org specified by the -u flag (json)', () => {
+    const username = env.getString('TESTKIT_HUB_USERNAME');
+    const json = execCmd(`auth:logout -p -u ${username} --json`, { ensureExitCode: 0 }).jsonOutput;
+    expect(json).to.deep.equal({
+      status: 0,
+      result: [username],
+    });
+  });
+
+  it('should remove the org specified by the -u flag (human readable)', () => {
+    const username = env.getString('TESTKIT_HUB_USERNAME');
+    const result = execCmd(`auth:logout -p -u ${username}`, { ensureExitCode: 0 });
+    const output = getString(result, 'shellOutput.stdout');
+    expect(output).to.equal(`Successfully logged out of orgs: ${username}\n`);
+  });
+
+  it('should fail if there is no default org and the -u flag is not specified (json)', () => {
+    const json = execCmd('auth:logout -p --json', { ensureExitCode: 1 }).jsonOutput as { name: string };
+    expect(json.name).to.equal('NoOrgFound');
+  });
+
+  it('should remove the default username if the -u flag is not specified (json)', () => {
+    const username = env.getString('TESTKIT_HUB_USERNAME');
+    execCmd(`config:set defaultusername=${username} --global`, { ensureExitCode: 0 });
```

Comment:
@mshanemc why would it be so bad to include config as dev plugin? It doesn't make any difference to users and it ensures that we have the commands we need during testing.

I like the dev plugin approach more than the `shelljs` approach you're proposing because I'd rather us _always_ use `execCmd` as opposed to sometimes doing one thing sometimes doing another thing

Reply:
OK, you've got me partially convinced.

My concerns were cross-dependencies (ex: two that end up depending on each other) and the experience we have now of (bump core, bump plugin, bump cli, lockfiles everywhere) and introducing extra steps to get everybody updated, especially if a plugin has a major version (or coreV3).

The "give me sfdx latest" seems easier.

Reply:
@mshanemc I think relying on an update to sfdx-cli@latest would be more frustrating than relying on an update to a plugin. For example, what happens if plugin-auth nut test requires a fix in plugin-config? If we rely on latest, then we have to wait for the change to be pulled in during the release which only happens once a week - and then what if there's a moratorium? Then you're blocked until the following week.

If we rely just on the plugin, we can bump the version as soon as the fix is merged in. I personally feel that that's easier

Reply:
Ya, I agree with Mike here. Also, it is only running on the dependency when targeting `./bin/run`. When the tests are run against `sfdx` (any version) it will use the config command from **that** CLI version which seems correct.